python/ray/serve/_private/deployment_graph_build.py

```diff
@@ -195,6 +199,9 @@ def replace_with_handle(node):
         ):
             deployment_name = deployment_shell.name
 
+        if app_name:
+            deployment_name = f"{app_name}_{deployment_name}"
```

Comment:
why is this optional? is it to support having a single "global" app? add to header comment please!